libagent/device/ledger.py

```diff
@@ -72,23 +92,42 @@ def pubkey(self, identity, ecdh=False):
     def sign(self, identity, blob):
         """Sign given blob and return the signature (as bytes)."""
         path = _expand_path(identity.get_bip32_address(ecdh=False))
-        if identity.identity_dict['proto'] == 'ssh':
-            ins = '04'
-            p1 = '00'
-        else:
-            ins = '08'
-            p1 = '00'
-        if identity.curve_name == 'nist256p1':
-            p2 = '81' if identity.identity_dict['proto'] == 'ssh' else '01'
-        else:
-            p2 = '82' if identity.identity_dict['proto'] == 'ssh' else '02'
-        apdu = '80' + ins + p1 + p2
-        apdu = binascii.unhexlify(apdu)
-        apdu += bytearray([len(blob) + len(path) + 1])
-        apdu += bytearray([len(path) // 4]) + path
-        apdu += blob
-        log.debug('apdu: %r', apdu)
-        result = bytearray(self.conn.exchange(bytes(apdu)))
+        offset = 0
+        result = None
+        while offset != len(blob):
+            data = bytes()
+            if offset == 0:
+                data += bytearray([len(path) // 4]) + path
+            if len(blob) - offset > (255-len(data)):
+                chunkSize = (255 - len(data))
+            else:
+                chunkSize = len(blob) - offset
+            data += blob[offset : offset + chunkSize]
+
+            if identity.identity_dict['proto'] == 'ssh':
+                ins = '04'
+            else:
+                ins = '08'
+
+            if identity.curve_name == 'nist256p1':
+                p2 = '81' if identity.identity_dict['proto'] == 'ssh' else '01'
+            else:
+                p2 = '82' if identity.identity_dict['proto'] == 'ssh' else '02'
+
+            if offset == 0:
+                p1 = "00"
+            elif offset + chunkSize == len(blob) and self.ledger_app_version_minor >= 8:
+                p1 = "81" # end of frame byte only handled in 0.0.8+
+            else:
+                p1 = "01"
+
+            apdu = binascii.unhexlify('80' + ins + p1 + p2) + len(data).to_bytes(1, 'little') + data
+
+            log.debug('apdu: %r', apdu)
+            result = bytearray(self.conn.exchange(bytes(apdu)))
```

Comment:
Not sure whether we should collect multiple responses into a single `bytearray`?

Reply:
you're right, i should check every answer anyways!

Reply:
actually, only the last one matters, since it contains the signed blob